cktap/proto.py

```diff
@@ -327,28 +344,37 @@ def get_slot_usage(self, slot, cvc=None):
 
         return (addr, status, here)
 
-    def sign_digest(self, cvc: str, digest: bytes, slot: int=0, subpath: str=None) -> bytes:
+    def sign_digest(self, cvc: str, digest: bytes, slot: int=0, subpath: str=None, bip32_path: str = None) -> bytes:
         """
         Sign 32 bytes digest and return 65 bytes long recoverable signature.
 
         Uses derivation path based on current set derivation on card plus optional
         subpath parameter which if provided, will be added to card derivation path.
         Subpath can only be of length 2 and non-hardened components only.
+        if subpath is specified - use current derivation + derive subpath
+        if bip32_path is specified - subpath is ignored and derivation goes from root
 
         Returns non-deterministic recoverable signature (header[1b], r[32b], s[32b])
         """
         if len(digest) != 32:
             raise ValueError("Digest must be exactly 32 bytes")
 
-        if not self.is_tapsigner and subpath:
-            raise ValueError("Cannot use 'subpath' option for SATSCARD")
+        if not self.is_tapsigner and (subpath or bip32_path):
+            raise ValueError("Cannot use 'subpath/bip32_path' option for SATSCARD")
+
+        if bip32_path:
+            # ignore subpath is bip32_path is provided
+            int_path = str2path(bip32_path) if bip32_path is not None else []
+            check_bip32_path(int_path)  # raises value error if hardened component after non-hardened
+            hardened, sub = split_bip32_path(int_path)
+            self.set_derivation(path=path2str(hardened), cvc=cvc)
```

Comment:
- this is slow; check if card already at correct prefix path and skip re-deriving it here
- also dislike path2str happening here; before for str2path to detect integers and become a no-op when not needed